### PR TITLE
added a few functions, cleaned up functions to use lro_wait. checking…

### PR DESCRIPTION
**Pull request by m-kovalsky**
… for if a model is the default semantic model, added readme reference to pyproject.toml

### CODE REVIEW

File: src/sempy_labs/_helper_functions.py [hunk 1]
```diff
@@ -506,3 +508,78 @@ def resolve_workspace_name_and_id(workspace: Optional[str] = None) -> Tuple[str,
...
+
+    workspace = fabric.resolve_workspace_name(workspace)
+
+    dfI = fabric.list_items(workspace=workspace)
+    dfI_filt = dfI[
+        (dfI["Display Name"] == dataset)
+        & (dfI["Type"].isin(["Lakehouse", "SemanticModel", "SQLEndpoint"]))
+    ]
+
+    if len(dfI_filt) == 3:
```
Comment (eisber) [inline src/sempy_labs/_helper_functions.py:548]: nit: you could also do return len(dfI_filt) == 3
up to you

File: src/sempy_labs/_list_functions.py [hunk 2]
```diff
@@ -2337,3 +2323,113 @@ def list_capacities() -> pd.DataFrame:
...
+            f"{icons.red_dot} The '{notebook_name}' notebook does not exist within the '{workspace}' workspace."
+        )
+
+    notebook_id = dfI_filt["Id"].iloc[0]
+    client = fabric.FabricRestClient()
+    response = client.post(
+        f"v1/workspaces/{workspace_id}/notebooks/{notebook_id}/getDefinition",
+        lro_wait=True,
+    )
+    df_items = pd.json_normalize(response.json()["definition"]["parts"])
```
Comment (eisber) [inline src/sempy_labs/_list_functions.py:2363]: check response status code

File: src/sempy_labs/_list_functions.py [hunk 3]
```diff
@@ -2337,3 +2323,113 @@ def list_capacities() -> pd.DataFrame:
...
+    client = fabric.FabricRestClient()
+    dfI = fabric.list_items(workspace=workspace, type="Notebook")
+    dfI_filt = dfI[dfI["Display Name"] == notebook_name]
+    if len(dfI_filt) > 0:
+        raise ValueError(
+            f"{icons.red_dot} The '{notebook_name}' already exists within the '{workspace}' workspace."
+        )
+
+    response = requests.get(url)
+    response.raise_for_status()
```
Comment (eisber) [inline src/sempy_labs/_list_functions.py:2408]: I'd suggest to keep the FabricHTTPException pattern

File: src/sempy_labs/report/_generate_report.py [hunk 4]
```diff
@@ -73,83 +69,52 @@ def create_report_from_reportjson(
...
+                    "payload": definitionPBIR,
+                    "payloadType": "InlineBase64",
+                },
+            ]
+        },
+    }
+
+    if theme_json is not None:
+        theme_payload = _conv_b64(theme_json)
+        theme_id = theme_json["payload"]["blob"]["displayName"]
```
Comment (eisber) [inline src/sempy_labs/report/_generate_report.py:95]: the displayName is used as id?

File: src/sempy_labs/report/_report_functions.py [hunk 5]
```diff
@@ -240,9 +240,9 @@ def export_report(
         )
 
     if file_name is None:
-        file_name = report + fileExt
+        file_name = f"{report}{fileExt}"
     else:
-        file_name = file_name + fileExt
+        file_name = f"{file_name}{fileExt}"
```
Comment (eisber) [inline src/sempy_labs/report/_report_functions.py:245]: nit: maybe do f"{file_name}.{file_ext}" and remove all the . from the dict? again, up to you.

**Review by eisber — APPROVED** [review]

File: src/sempy_labs/_helper_functions.py (hunk 1 above)
Comment (m-kovalsky) [reply src/sempy_labs/_helper_functions.py:548]: nice, more elegant. i like it.

File: src/sempy_labs/report/_generate_report.py (hunk 4 above)
Comment (m-kovalsky) [reply src/sempy_labs/report/_generate_report.py:95]: it matches to the name of the .json file

File: src/sempy_labs/report/_report_functions.py (hunk 5 above)
Comment (m-kovalsky) [reply src/sempy_labs/report/_report_functions.py:245]: think it's ok

File: src/sempy_labs/_query_scale_out.py [hunk 6]
```diff
@@ -102,25 +102,25 @@ def qso_sync_status(
...
-            "Sync Start Time": o["syncStartTime"],
-            "Sync End Time": o["syncEndTime"],
-            "Commit Version": o["commitVersion"],
-            "Commit Timestamp": o["commitTimestamp"],
-            "Target Sync Version": o["targetSyncVersion"],
-            "Target Sync Timestamp": o["targetSyncTimestamp"],
-            "Trigger Reason": o["triggerReason"],
-            "Min Active Read Version": o["minActiveReadVersion"],
-            "Min Active Read Timestamp": o["minActiveReadTimestamp"],
+            "Scale Out Status": o.get("scaleOutStatus", {}),
```
Comment (eisber) [inline src/sempy_labs/_query_scale_out.py:105]: why is the default {} ? we only used that if there are subsequent lookups. I'd guess if scaleOutStatus isn't there, you just want None. so o.get("scaleOutStatus")?

**Review by eisber — CHANGES_REQUESTED** [review]

**Pull request merged**